packages/hardhat-core/src/internal/hardhat-network/provider/transactions/FakeSenderTransaction.ts

```diff
@@ -31,6 +31,12 @@ export class FakeSenderTransaction extends Transaction {
     serialized: Buffer,
     opts?: TxOptions
   ): never {
+    const values = rlp.decode(serialized);
+
+    if (!Array.isArray(values)) {
+      throw new Error("Invalid serialized tx input. Must be array");
+    }
+
```

Comment:
What's the point of this check if the method throws a "not implemented" later anyway?

Reply:
Ouch

Reply:
deleted